common/src/main/java/org/jboss/gm/common/utils/OTELUtils.java

```diff
@@ -0,0 +1,51 @@
+package org.jboss.gm.common.utils;
+
+import lombok.experimental.UtilityClass;
+
+import org.gradle.api.logging.LogLevel;
+import org.gradle.api.logging.Logger;
+import org.jboss.gm.common.logging.FilteringCustomLogger;
+import org.jboss.gm.common.logging.GMLogger;
+
+import com.redhat.resilience.otel.OTelCLIHelper;
+
+@UtilityClass
+public class OTELUtils {
+
+    private final Logger logger = GMLogger.getLogger(OTELUtils.class);
+
+    private final LogLevel originalLevel = FilteringCustomLogger.getContext().getLevel();
+
+    public void startOTel() {
+        String endpoint = System.getenv("OTEL_EXPORTER_OTLP_ENDPOINT");
+        String service = System.getenv("OTEL_SERVICE_NAME");
+
+        if (endpoint != null) {
+            if (service == null) {
+                service = "gradle-manipulator";
+            }
+            logger.info("Enabling OpenTelemetry collection on {} with service name {} level {}", endpoint, service,
+                    originalLevel);
+            try {
+                if (originalLevel != LogLevel.DEBUG) {
+                    FilteringCustomLogger.getContext().setLevel(LogLevel.DEBUG);
+                }
+                OTelCLIHelper.startOTel(service, "alignment-plugin",
+                        OTelCLIHelper.defaultSpanProcessor(OTelCLIHelper.defaultSpanExporter(endpoint)));
+            } finally {
+                FilteringCustomLogger.getContext().setLevel(originalLevel);
+            }
+        }
+    }
+
+    public void stopOTel() {
+        try {
+            if (originalLevel != LogLevel.DEBUG) {
+                FilteringCustomLogger.getContext().setLevel(LogLevel.DEBUG);
```

Comment:
Should be `originalLevel`?

Reply:
Eh, no, you do set that in the finally. But, it should have already been set on start, line 36, so I wasn't sure.

Reply:
In each start/stop there is a wrapper around the logging control.